### PR TITLE
Code missing from master branch

### PR DESCRIPTION
**Pull request by symphonyeyc**
### Contributing guide

https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md
### Changelog

*
*
*
*
*

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@symphonyeyc__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
__@symphonyeyc__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

File: 1-CONTRIBUTION-GUIDE/best-practices.md [hunk 1]
```diff
@@ -18,13 +18,12 @@ You are currently reading the best practices.
...
- + Things that are globally unique (e.g. website name).  These are usually endpoints that the user may need to be aware of, however in many cases a unique name can be generated automatically.
+1. Minimize parameters whenever possible, this allows for a good "hello world" experience where the user doesn't have to answer a number of questions to complete a deployment.  If you can use a variable or a literal, do so.  Users who want to parameterize something will likely have the skills to do so. Only provide parameters for:
+ + Things that are globally unique (e.g. website name).  These are usually endpoints that the user may need to be aware of, however in many cases a unique name can be generated automatically by using the [uniqueString()](https://azure.microsoft.com/en-us/documentation/articles/resource-group-template-functions/#uniquestring) template language function.
```
Comment (bmoore-msft) [inline 1-CONTRIBUTION-GUIDE/best-practices.md]: Add another bullet that suggests using an output for generated names that are endpoints.  E.g. if I generate a website (or VM) name, add an output to the template that outputs the uri or fqdn that was generated).

    "outputs": {
    "vm-fqdn": {
      "type": "string",
      "value": "[reference(variables('publicIPAddressName')).dnsSettings.fqdn]"
    }
  }

File: 1-CONTRIBUTION-GUIDE/best-practices.md [hunk 2]
```diff
@@ -53,27 +52,38 @@ The following guidelines are relevant to the main deployment template and nested
    }
  ]
  ```
+ + If you must use a location parameter, share parameters whenever possible - e.g. the location parameter should be shared among resources that must or are likely to be in the same location.
 
-5. Name **variables** using this scheme **templateScenarioResourceName** (e.g. simpleLinuxVMVNET, userRoutesNSG, elasticsearchPublicIP etc.) that describe the scenario rather. This ensures when a user browses all the resources in the Portal there aren't a bunch of resources with the same name (e.g. myVNET, myPublicIP, myNSG)
-6. For many resources with a resource group, a name is not often relevant and using something like "storageAccount" may be acceptable.  You can also use variables for the name of a resource. Use **displayName** tags for a "friendly" name in the JSON outline view.  This should ideally match the name property value or property name.
+1. Name **variables** using this scheme **templateScenarioResourceName** (e.g. simpleLinuxVMVNET, userRoutesNSG, elasticsearchPublicIP etc.) that describe the scenario rather. This ensures when a user browses all the resources in the Portal there aren't a bunch of resources with the same name (e.g. myVNET, myPublicIP, myNSG)
```
Comment (bmoore-msft) [inline 1-CONTRIBUTION-GUIDE/best-practices.md]: Reword this?  I though this mean the name of the variable itself... not it's value (when applied to the resource)...  e.g.

Variables that are used for resource names should use the following pattern ..........

File: 1-CONTRIBUTION-GUIDE/best-practices.md [hunk 3]
```diff
@@ -214,15 +214,80 @@ It is obvious to create a single deployment template for deploying a single reso
...
+              "description": "The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated."
+          },
+          "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-custom-script-windows/"
+      },
+      "_artifactsLocationSasToken": {
+          "type": "securestring",
+          "metadata": {
+              "description": "The sasToken required to access _artifactsLocation.  When the template is deployed using the accompanying scripts, a sasToken will be automatically generated."
+          },
+          "defaultValue": "?"
```
Comment (bmoore-msft) [inline 1-CONTRIBUTION-GUIDE/best-practices.md]: change this to empty string, the portal has a problem with "?"

**Comment by bmoore-msft** [comment]
These are excellent changes - thanks for the update...

**Pull request closed without merging**